eventActions.js

```diff
@@ -511,11 +537,19 @@ ActionSetProp.prototype.setPointers = function (entitiesArr) {
  * @param {ko.observableArray} entitiesArr - this is the knockout array that holds all instances.
  */
 ActionSetProp.prototype.reAddEntities = function (entitiesArr) {
-    if (this.operand && this.operand.reAddEntities) {
-        this.operand.reAddEntities(entitiesArr);
+    var refToObjectProperty = this.refToObjectProperty();
+    for (var i = 0; i < refToObjectProperty.length; i++) {
+        var variable = refToObjectProperty[i].variable;
+        if (variable) {
+            if (!entitiesArr.byId.hasOwnProperty(variable.id())) {
+                entitiesArr.push(variable);
+            }
+        }
     }
-    if (this.refToObjectProperty && this.refToObjectProperty.reAddEntities) {
-        this.refToObjectProperty.reAddEntities(entitiesArr);
+    for (var i = 0; i < this.operand().length; i++) {
```

Comment:
for performance, it would be good to first only read the observableArray once... like you do above with refToObjectProperty...
Because always again in the loop the access with () takes time, so it is better to do this only one time before the loop.

Reply:
Done